llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -2430,6 +2430,11 @@ void SpirvLowerGlobal::lowerBufferBlock() {
 
               bool isNonUniform = false;
 
+              bool forceNonUniform = isShaderStageInMask(
+                  m_shaderStage,
+                  m_context->getPipelineContext()->getPipelineOptions()->forceNonUniformResourceIndexStageMask);
+              isNonUniform |= forceNonUniform;
+
               // Run the users of the GEP to check for any nonuniform calls.
               for (User *const user : getElemPtr->users()) {
```

Comment:
I think you need to surround this `for` with an `if (!isNonUniform)`, otherwise `isNonUniform` may be overwritten with `false` in the loop.